clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -224,6 +230,8 @@ public synchronized boolean containsTopic(String topic) {
      */
     public synchronized void update(Cluster newCluster, Set<String> unavailableTopics, long now) {
         Objects.requireNonNull(newCluster, "cluster should not be null");
+        if (isClosed())
+            throw new IllegalStateException("Update requested after metadata close");
```

Comment:
Should we use `KafkaException` here as well?

Reply:
Hmm, I feel `IllegalStateException` is more appropriate in this case. We expect `NetworkClient` to not invoke `update` after it has called `Metadata#close()`.